cli/permissions.rs

```diff
@@ -288,6 +288,109 @@ impl DenoPermissions {
       n => Err(OpError::other(format!("No such permission name: {}", n))),
     }
   }
+
+  #[allow(clippy::too_many_arguments)]
+  fn _fork(
+    &self,
+    allow_read: bool,
+    read_whitelist: HashSet<PathBuf>,
+    allow_write: bool,
+    write_whitelist: HashSet<PathBuf>,
+    allow_net: bool,
+    net_whitelist: HashSet<String>,
+    allow_env: bool,
+    allow_run: bool,
+    allow_plugin: bool,
+    allow_hrtime: bool,
+  ) -> Result<DenoPermissions, OpError> {
+    if allow_read && self.allow_read == PermissionState::Deny {
```

Comment:
How about adding `fork` method to `PermissionState`? `self.allow_read.fork(allow_read)?` - then you can encapsulate check and returning error in a single function